### PR TITLE
Improve Assert, BLE, Buffer, Dgram docs

### PR DESCRIPTION
**Pull request by hs0225**
IoT.js-DCO-1.0-Signed-off-by: Hosung Kim hs852.kim@samsung.com

### CODE REVIEW

File: docs/api/IoT.js-API-Buffer.md [hunk 1]
```diff
@@ -449,10 +471,10 @@ print(buffer); // prints .DCBA.
...
 
 
-### `buf.readInt8(offset[, noAssert])` <a name="buf-read-int8"></a>
-* `offset: Number` - start position of buffer for reading.
-* `noAssert: Boolean`, Default: false - skip offset validation.
-* Return: `Number`
+### `buf.readInt8(offset[, noAssert])`
+* `offset <Number>` - start position of buffer for reading.
+* `noAssert <Boolean>` Default: false - skip offset validation.
+* Returns: `Number`
```
Comment (chokobole) [inline docs/api/IoT.js-API-Buffer.md]: Please wrap type with `<>`

File: docs/api/IoT.js-API-Buffer.md [hunk 2]
```diff
@@ -468,10 +490,10 @@ print(buffer.readUInt8(1).toString(16)); // prints 42
...
 
 
-### `buf.readUInt8(offset[, noAssert])` <a name="buf-read-uint8"></a>
-* `offset <Integer>`, start position of the reading
-* `noAssert <Boolean>`, skip argument validation. Default: `false`
-* Return: `Number`
+### `buf.readUInt8(offset[, noAssert])`
+* `offset <Integer>` - start position of the reading
+* `noAssert <Boolean>` Default: `false` - skip argument validation
+* Returns: `Number`
```
Comment (chokobole) [inline docs/api/IoT.js-API-Buffer.md]: ditto

File: docs/api/IoT.js-API-Buffer.md [hunk 3]
```diff
@@ -487,10 +509,10 @@ print(buffer.readUInt8(1).toString(16)); // prints 42
...
 
 
-### `buf.readUInt16LE(offset[, noAssert])` <a name="buf-read-uint16"></a>
-* `offset: Number` - start position of buffer for reading.
-* `noAssert: Boolean`, Default: false - skip offset validation.
-* Return: `Number`
+### `buf.readUInt16LE(offset[, noAssert])`
+* `offset <Number>` - start position of buffer for reading.
+* `noAssert <Boolean>` Default: false - skip offset validation.
+* Returns: `Number`
```
Comment (chokobole) [inline docs/api/IoT.js-API-Buffer.md]: ditto

File: docs/api/IoT.js-API-Buffer.md [hunk 4]
```diff
@@ -487,10 +509,10 @@ print(buffer.readUInt8(1).toString(16)); // prints 42
 ```
 
 
-### `buf.readUInt16LE(offset[, noAssert])` <a name="buf-read-uint16"></a>
-* `offset: Number` - start position of buffer for reading.
-* `noAssert: Boolean`, Default: false - skip offset validation.
-* Return: `Number`
+### `buf.readUInt16LE(offset[, noAssert])`
+* `offset <Number>` - start position of buffer for reading.
+* `noAssert <Boolean>` Default: false - skip offset validation.
```
Comment (chokobole) [inline docs/api/IoT.js-API-Buffer.md]: Please wrap false with ``

File: docs/api/IoT.js-API-DGRAM.md [hunk 5]
```diff
@@ -18,157 +16,254 @@ The following shows dgram module APIs available for each platform.
...
-* `sendListener: Function(err, length)`
- * `err: Null | Error`
-   * `code: String`, temporally, it is just a String of *"error"*.
-   * `errno: String`, it is same with `code`
-   * `syscall: Number`
-   * `address: String`
-   * `port: Number`
- * `length: Number`, it indicates the length of data.
+### `socket.send(msg, [offset, length], port, address[, sendListener])`
+* `msg <Buffer | String | Array>`
```
Comment (chokobole) [inline docs/api/IoT.js-API-DGRAM.md]: Is it proper to type this way `<Buffer>|<String>|<Array>`?

File: docs/api/IoT.js-API-Buffer.md [hunk 6]
```diff
@@ -449,10 +471,10 @@ print(buffer); // prints .DCBA.
 ```
 
 
-### `buf.readInt8(offset[, noAssert])` <a name="buf-read-int8"></a>
-* `offset: Number` - start position of buffer for reading.
-* `noAssert: Boolean`, Default: false - skip offset validation.
-* Return: `Number`
+### `buf.readInt8(offset[, noAssert])`
+* `offset <Number>` - start position of buffer for reading.
+* `noAssert <Boolean>` Default: false - skip offset validation.
```
Comment (chokobole) [inline docs/api/IoT.js-API-Buffer.md]: Wrap false with ``

File: docs/api/IoT.js-API-DGRAM.md [hunk 7]
```diff
@@ -18,157 +16,254 @@ The following shows dgram module APIs available for each platform.
...
- * `rinfo: Object`
-   * `address: String`
-   * `family: String`: it indicates an address family either `IPv4` or `IPv6`
-   * `port: Number`
+### `dgram.createSocket(options[, createListener])`
+### `dgram.createSocket(type[, createListener])`
+* `options <Object>`
+  * `type <String>` - it indicates an address family either `udp4` or `udp6`.
+  * `reuseAddr <Boolean>` - it indicates to allow socket to bind the address and port used previously.
+* `type <String>` - it indicates an address family either 'udp4' or 'udp6'.
```
Comment (chokobole) [inline docs/api/IoT.js-API-DGRAM.md]: Are these udp4 and udp6 also wrapped with ``?

**Comment by hs0225** [comment]
@chokobole I updated docs. Please review again.

**Comment by chokobole** [comment]
LGTM

**Pull request merged**